cli/pcluster/config/update_policy.py

```diff
@@ -0,0 +1,129 @@
+                fail_reason = self.fail_reason
+                action_needed = self.action_needed
+        else:
+            result = UpdatePolicy.CheckResult.FAILED
+            fail_reason = self.fail_reason
+            action_needed = self.action_needed
+
+        if callable(action_needed):
+            action_needed = action_needed(change, patch)
+
+        if callable(fail_reason):
+            fail_reason = fail_reason(change, patch)
+
+        return result, fail_reason, action_needed
+
+    def __eq__(self, other):
+        if not isinstance(other, UpdatePolicy):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
+
+        return self.fail_reason == other.fail_reason and self.level == other.level
+
+
+# Base policies
+UpdatePolicy.IGNORED = UpdatePolicy(level=-10, fail_reason="-", condition_checker=(lambda change, patch: True))
+UpdatePolicy.SUPPORTED = UpdatePolicy(
+    level=0, fail_reason="-", condition_checker=(lambda change, patch: True)
+)  # Can be safely updated
+UpdatePolicy.AWSBATCH_CE_MAX_RESIZE = UpdatePolicy(
+    level=1,
+    fail_reason=lambda change, patch: "Max vCPUs is lower than current Desired vCPUs ({0})".format(
+        utils.get_batch_ce_capacity(patch.stack_name)
+    ),
+    action_needed="Stop the cluster with the following command:\npcluster stop -c {config_file} {cluster_name}",
+    condition_checker=lambda change, patch: utils.get_batch_ce_capacity(patch.stack_name)
+    <= patch.target_config.get_section("cluster").get_param_value("max_vcpus"),
+)  # Checks resize of max_vcpus in Batch Compute Environment
+UpdatePolicy.INCREASE_ONLY = UpdatePolicy(
+    level=2,
+    fail_reason=lambda change, patch: "Value of parameter {0} cannot be decreased".format(change.param_key),
+    action_needed=lambda change, patch: "Set the value of parameter {0} to {1} or greater".format(
+        change.param_key, change.old_value
+    ),
+    condition_checker=lambda change, patch: change.new_value >= change.old_value,
+)  # Checks that the value of the parameter has not been decreased
+UpdatePolicy.COMPUTE_FLEET_STOP = UpdatePolicy(
+    level=10,
+    fail_reason="Compute fleet must be empty",
+    action_needed="Stop the cluster with the following command:\npcluster stop -c {config_file} {cluster_name}",
+    condition_checker=lambda change, patch: len(utils.get_asg_instances(patch.stack_name)) == 0,
+)  # Can be updated but compute fleet must be restarted
+UpdatePolicy.MASTER_STOP = UpdatePolicy(
+    level=20,
+    fail_reason="Master node must be down",
+    action_needed="Stop the cluster with the following command:\npcluster stop -c {config_file} {cluster_name}",
+    condition_checker=lambda change, patch: utils.get_master_server_state(patch.stack_name) == "stopped",
+)  # Can be updated but master node must be restarted
+UpdatePolicy.UNKNOWN = UpdatePolicy(
+    level=100,
+    fail_reason="Update currently not supported",
```

Comment:
UNKNOWN changes must never appear in the report. All policies must have their own specific update logic associated (there is also a unit test to ensure this is happening) and in any case an unknown change would be treated as UNSUPPORTED. So I'd suggest keeping a different wording here to let us detect an UNKNOWN policy erroneously being displayed. Do you have any suggestion? Maybe just removing/replacing the word "currently"?

Reply:
I'd like keeping it because it works like a marker for parameters not yet being linked to a specific update policy. A specific unit test has been added to find out those parameters. If we'd set new parameters by default to ``UNSUPPORTED`` we'd miss this feature.